ui/tx/Noves/utils/NovesGetSplittedDescription.ts

```diff
@@ -0,0 +1,180 @@
+import _ from 'lodash';
+
+import type { NovesResponseData } from 'types/novesApi';
+
+interface TokensData {
+  nameList: Array<string>;
+  symbolList: Array<string>;
+  idList: Array<string>;
+  names: {
+    [x: string]: {
+      name: string | undefined;
+      symbol: string | undefined;
+      address: string | undefined;
+      id?: string | undefined;
+    };
+  };
+  symbols: {
+    [x: string]: {
+      name: string | undefined;
+      symbol: string | undefined;
+      address: string | undefined;
+      id: string | undefined;
+    };
+  };
+}
+
+function getTokensData(data: NovesResponseData): TokensData {
+  const sent = data.classificationData.sent || [];
+  const received = data.classificationData.received || [];
+
+  const txItems = [ ...sent, ...received ];
+
+  const tokens = txItems.map((item) => {
+    const name = item.nft?.name || item.token?.name;
+    const symbol = item.nft?.symbol || item.token?.symbol;
+
+    const token = {
+      name: name,
+      symbol: symbol?.toLowerCase() === name?.toLowerCase() ? undefined : symbol,
+      address: item.nft?.address || item.token?.address,
+      id: item.nft?.id || item.token?.id,
+    };
+
+    return token;
+  });
+
+  const tokensGroupByname = _.groupBy(tokens, 'name');
+  const tokensGroupBySymbol = _.groupBy(tokens, 'symbol');
+  const tokensGroupById = _.groupBy(tokens, 'id');
+
+  const mappedNames = _.mapValues(tokensGroupByname, (i) => {
+    return i[0];
+  });
+
+  const mappedSymbols = _.mapValues(tokensGroupBySymbol, (i) => {
+    return i[0];
+  });
+
+  const mappedIds = _.mapValues(tokensGroupById, (i) => {
+    return i[0];
+  });
+
+  const nameList = _.keysIn(mappedNames).filter(i => i !== 'undefined');
+  const symbolList = _.keysIn(mappedSymbols).filter(i => i !== 'undefined');
+  const idList = _.keysIn(mappedIds).filter(i => i !== 'undefined');
+
+  return {
+    nameList,
+    symbolList,
+    idList,
+    names: mappedNames,
+    symbols: mappedSymbols,
+  };
+}
+
+export const NovesGetSplittedDescription = (translateData: NovesResponseData) => {
```

Comment:
Personally, I find it challenging to reason about this function. The logic is complex and heavy. Do you think it's possible to split this function into separate, independent, and clear functions?